### DISCUSSION TITLE
Worked with JK-B2A24S20P (Hw: 8.X, Sw: 8.20G) and ESP8266, problems with ESP32

### DISCUSSION THREAD

**Discussion by helopez** [syssi/esphome-jk-bms#53] [Show and tell]
I got it working with a Wemos D1 mini, had to change serial pins to GPIO15 (tx) and GPIO13 (rx). Didn't work with the default pins GPIO1 and GPIO3, I guess those pins conflicted with the integrated usb to serial.

Hardware revision: 8.X
Software revision: 8.20G

I tried using two different ESP32-DevKitC, a WROOM and a WROVER. Both of them gave me a stack overflow error that I couldn't find, luckily I got it working with the ESP8266.

Error:

ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:10124
load:0x40080400,len:5828
entry 0x400806a8
***ERROR*** A stack overflow in task loopTask has been detected.
abort() was called at PC 0x40088b98 on core 1

ELF file SHA256: 0000000000000000

Backtrace: 0x40088904:0x3ffafd60 0x40088b81:0x3ffafd80 0x40088b98:0x3ffafda0 0x4008a77b:0x3ffafdc0 0x4008c344:0x3ffafde0 0x4008c2fa:0xa71a3601

Rebooting...

Thanks to the developers and contributors

**Comment by syssi** [comment]
The stack overflow at the ESP32 happens because of the large number of sensors. Please reduce the sensors to just a few and try again. This time you shouldn't see the stack overflow error. If everything is working please decrease the log level to be less verbose and increase the number of sensors step by step.

**Reply by helopez** [reply]
Thanks for the suggestion, great work!

Update: I removed some sensors and the ESP32 now works, thank you.
